### PR TITLE
Fixed and optimized the executor

### PR DESCRIPTION
**Pull request by 02p01r**
TBD

### CODE REVIEW

File: src/dag/dag.cpp [hunk 1]
```diff
@@ -494,6 +494,9 @@ std::pair<uint64_t, std::shared_ptr<vec_blk_t>> DagManager::getDagBlockOrder(blk
 
 uint DagManager::setDagBlockOrder(blk_hash_t const &new_anchor, uint64_t period, vec_blk_t const &dag_order,
                                   const taraxa::DbStorage::BatchPtr &write_batch) {
+  // TODO this function smells. It tries to manage in-memory and persistent state at the same time, which it
```
Comment (justinsnapp) [inline src/dag/dag.cpp:497]: Yes this has long been a stinky part of our code that we have never refactored.

**Review by justinsnapp — APPROVED** [review]

File: src/consensus/pbft_manager.cpp [hunk 2]
```diff
@@ -1232,13 +1232,23 @@ bool PbftManager::pushPbftBlock_(PbftBlockCert const &pbft_block_cert_votes) {
...
+  }
+
   // Commit DB
   db_->commitWriteBatch(batch);
   LOG(log_nf_) << node_addr_ << " successful push unexecuted pbft block " << pbft_block_hash << " in period "
                << pbft_period << " into chain! In round " << getPbftRound();
 
-  // Notify executor
-  executor_->cv_executor.notify_one();
+  executor_->execute(pbft_block);
```
Comment (JakubFornadel) [inline src/consensus/pbft_manager.cpp:1252]: What would happen if i call execute(block) with block that was already processed ?

**Comment by JakubFornadel** [comment]
What is the problem/bug that this PR is supposed to fix ?

File: src/node/executor.cpp [hunk 3]
```diff
@@ -17,171 +17,189 @@ Executor::Executor(addr_t node_addr, std::shared_ptr<DbStorage> db, std::shared_
       db_(db),
       dag_mgr_(dag_mgr),
       trx_mgr_(trx_mgr),
-      final_chain_(final_chain),
-      pbft_chain_(pbft_chain) {
+      final_chain_(final_chain) {
```
Comment (JakubFornadel) [inline src/node/executor.cpp:20]: Why was pbft_chain_(pbft_chain) deleted ? Does not seem ok

**Pull request merged**